src/tools/SimpleTraversability.cpp

```diff
@@ -0,0 +1,253 @@
+                }
+            }
+        }
+    }
+
+    // Perform some post processing if required.
+    if (config.minPassageWidth > 0)
+    {
+        closeNarrowPassages(traversabilityOut, config.minPassageWidth);
+    }
+
+    if (config.obstacleClearance > 0)
+    {
+        growObstacles(traversabilityOut, config.obstacleClearance);
+    }
+
+    // Registers TraversabilityClasses in the TraversabilityGrid.
+    traversabilityOut.setTraversabilityClass(CLASS_OBSTACLE, grid::TraversabilityClass(0));
+
+    float driveability = 0.0;
+    for (int i = 0; i < config.classCount; ++i)
+    {
+        driveability = (1.0 / config.classCount) * (i + 1);
+        traversabilityOut.setTraversabilityClass(CUSTOM_CLASSES + i, grid::TraversabilityClass(driveability));
+    }
+
+    // Calculates the mean traversability class of the current map ignoring unknown areas.
+    uint8_t classId = 0;
+    double summedClassIds = 0.0;
+    double counter = 0.0;
+    for (int y = 0; y < height; ++y)
+    {
+        for (int x = 0; x < width; ++x)
+        {
+            classId = traversabilityOut.getTraversabilityClassId(x, y);
+
+            // Sum up if not unknown.
+            if(classId != CLASS_UNKNOWN) {
+                summedClassIds += classId;
+                counter++;
+            }
+        }
+    }
+
+    double medianDriveability = 0.55;
+    uint8_t meanClassId = counter == 0 ? 0 : summedClassIds / counter + 0.5;
+
+    // If the map is completely unkown or full with obstacles,
+    // set the driveability of unknown areas to the median driveability (0.55).
+    if(meanClassId <= 1) {
+        traversabilityOut.setTraversabilityClass(CLASS_UNKNOWN, grid::TraversabilityClass(medianDriveability));
+    } else {
+        grid::TraversabilityClass meanTravClass = traversabilityOut.getTraversabilityClass(meanClassId);
+        traversabilityOut.setTraversabilityClass(CLASS_UNKNOWN, meanTravClass);
+    }
+
+    return true;
+}
+
+void SimpleTraversability::closeNarrowPassages(grid::TraversabilityGrid& traversabilityGridIn, double minPassageWidth)
+{
+    SimpleTraversabilityRadialLUT lut;
+    lut.precompute(minPassageWidth, traversabilityGridIn.getResolution()[0], traversabilityGridIn.getResolution()[1]);
+
+    size_t mapWidth = traversabilityGridIn.getNumCells()[0];
+    size_t mapHeight = traversabilityGridIn.getNumCells()[1];
+
+    grid::TraversabilityGrid traversabilityGridOut = traversabilityGridIn;
+
+    for (size_t y = 0; y < mapHeight; ++y)
+    {
+        for (size_t x = 0; x < mapWidth; ++x)
+        {
+            int traversabilityClassId = traversabilityGridIn.at(grid::Index(x, y)).getTraversabilityClassId();
+            if (traversabilityClassId == CLASS_OBSTACLE)
+            {
+                lut.markAllRadius(traversabilityGridOut, x, y, CLASS_OBSTACLE);
+            }
+        }
+    }
+
+    std::swap(traversabilityGridOut, traversabilityGridIn);
```

Comment:
The swap seems to be unnecessary. What about copying the grid temporary to look up the traversability class and directly modifying the input grid?
The same can be done in the growObstacles method.

Reply:
I agree, that would be better. I will make the change.